### PR TITLE
Add Log for Fast Object Loader Failure

### PR DESCRIPTION
**Pull request by annym**
If the FoL fails we have no logs that provide data on the actual error.

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java [hunk 1]
```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
...
+                            log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
+                                    Utils.toReadableId(entry.getKey()));
+                            return;
+                        }
+
+                        // For now one by one read and apply
+                        for (long address : checkPoint.getAddresses()) {
+                            updateCorfuObject(getLogData(runtime, loadInCache, address));
+                        }
+                    } catch (Throwable t) {
```
Comment (PavelZaytsev) [inline runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: Throwable is very broad, maybe we can be more specific?

Comment (annym) [reply runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: The thing here is that if anything goes wrong at this level (exception, error, whatever) we want to log it before this completes exceptionally. For instance we were seeing StackOverflowError which wouldn't have been captured if we caught any more specific exception.

File: runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java [hunk 2]
```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
...
+                        if (checkPoint != null) {
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: I think `checkPoint` will not be null here: `CheckPoint checkPoint = entry.getValue().getLatestCheckPoint();` This is a simple operation which should not throw any exception, then the above code checks `if (checkPoint == null)`, which will return. So if we get a Throwable here, it means checkpoint is not null. So I think you can remove this null check.

File: runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java [hunk 3]
```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
...
-                    // For now one by one read and apply
-                    for (long address : checkPoint.getAddresses()) {
-                        updateCorfuObject(getLogData(runtime, loadInCache, address));
-                    }
-                }, executorService);
+                    CheckPoint checkPoint = null;
+                    try {
+                        checkPoint = entry.getValue().getLatestCheckPoint();
+                        if (checkPoint == null) {
+                            log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
```
Comment (hisundar) [inline runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: would logging the UUID also help here?

File: runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java [hunk 4]
```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
...
+
+                        // For now one by one read and apply
+                        for (long address : checkPoint.getAddresses()) {
+                            updateCorfuObject(getLogData(runtime, loadInCache, address));
+                        }
+                    } catch (Throwable t) {
+                        if (checkPoint != null) {
+                            log.error("resurrectCheckpoints: error on addresses {}", checkPoint.getAddresses());
+                        }
+                        log.error("resurrectCheckpoints: ", t);
```
Comment (hisundar) [inline runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: will logging UUID help here?

File: runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java (hunk 2 above)
Comment (annym) [reply runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java]: Done.

**Pull request merged**